# 02 — Exploratory Data Analysis

## Australian Energy Market Analytics

This notebook explores temporal patterns in electricity demand and spot prices
using Australian Energy Market Operator (AEMO) data.

### Objectives

- Examine the distribution of electricity demand and spot prices
- Identify daily and intraday demand patterns
- Compare weekday and weekend electricity consumption
- Investigate spot-price volatility
- Identify extreme and negative price events
- Develop hypotheses for subsequent demand-price analysis

## 1. Setup

In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

df = pd.read_csv(
    "../data/processed/clean_price_demand.csv",
    parse_dates=["SETTLEMENTDATE"]
)

df.head()

,REGION,SETTLEMENTDATE,TOTALDEMAND,RRP,PERIODTYPE
0,NSW1,2026-08-01 00:05:00,8898.91,89.88,TRADE
1,NSW1,2026-08-01 00:10:00,9032.33,95.05,TRADE
2,NSW1,2026-08-01 00:15:00,8989.12,88.88,TRADE
3,NSW1,2026-08-01 00:20:00,9036.72,88.88,TRADE
4,NSW1,2026-08-01 00:25:00,8946.67,84.79,TRADE


## 2. Descriptive Statistics

This section provides an initial overview of electricity demand and spot-price
behaviour before examining their temporal patterns.

In [5]:
df[["TOTALDEMAND", "RRP"]].describe()

,TOTALDEMAND,RRP
count,7776.000000,7776.000000
mean,8210.862065,77.059875
std,1295.783303,36.904946
min,4628.030000,-18.510000
25%,7297.312500,57.390000
50%,8129.005000,73.535000
75%,9112.190000,97.630000
max,11651.750000,311.910000


### Interpretation

The dataset contains 7,776 five-minute observations of electricity demand and
spot prices.

Electricity demand averaged approximately **8,210.86 MW**, with a median of
**8,129.01 MW**. The relatively small difference between the mean and median
suggests that demand is reasonably centred around 8,000 MW over the observed
period. Demand ranged from **4,628.03 MW** to **11,651.75 MW**, indicating
substantial variation in electricity consumption across different times of
the day and days of the month.

The average Regional Reference Price (RRP) was approximately **77.06/MWh**,
while the median was slightly lower at **73.54/MWh**. Prices ranged from
**-18.51/MWh**  to  **311.91/MWh**, demonstrating considerably greater
volatility than electricity demand. The negative minimum price is particularly
notable, as it indicates that electricity prices fell below zero during at
least one five-minute interval.

The relatively high maximum RRP compared with the upper quartile
(**$97.63/MWh**) also suggests the presence of occasional high-price events.
These observations motivate further investigation into the temporal patterns
of demand, negative prices, and price spikes.

### Relative Variability

To compare the variability of electricity demand and spot prices on a common
scale, the coefficient of variation (CV) is calculated as:

$$
CV = \frac{\sigma}{\mu} \times 100\%
$$

Electricity demand has a coefficient of variation of approximately **15.8%**,
while the Regional Reference Price (RRP) has a substantially higher coefficient
of variation of approximately **47.9%**.

This indicates that **spot prices are considerably more volatile relative to
their average level than electricity demand**. While electricity demand shows
moderate variation around its mean, spot prices experience much larger relative
fluctuations, suggesting the presence of periods of elevated price volatility.

This motivates further investigation into the timing and magnitude of electricity
price fluctuations in the subsequent analysis.

In [6]:
demand_cv = df["TOTALDEMAND"].std() / df["TOTALDEMAND"].mean() * 100
price_cv = df["RRP"].std() / df["RRP"].mean() * 100

print(f"Demand coefficient of variation: {demand_cv:.2f}%")
print(f"Price coefficient of variation: {price_cv:.2f}%")

Demand coefficient of variation: 15.78%
Price coefficient of variation: 47.89%
